java/flight/flight-core/src/test/java/org/apache/arrow/flight/TestDictionaryUtils.java

```diff
@@ -78,7 +79,7 @@ public void testCreateSchema() {
       Schema newSchema = DictionaryUtils.generateSchema(schema, dictProvider, dictionaryUsed);
 
       // assert that a new schema is created.
-      assertTrue(schema != newSchema);
+      assertNotEquals(schema, newSchema);
```

Comment:
Just curious, if we use assertNotSame, it would mean that it is not the same object in memory. But in this case, I thought we are merely checking the status of the created objects. Please correct me if I am wrong.

Reply:
Look at the comment; this is explicitly about a new object